src/formatDate.js

```diff
@@ -9,6 +9,40 @@
  */
 function formatDate(date, fromFormat, toFormat) {
   // write code here
+  const dateObj = {};
+  const splitDate = date.split(fromFormat[fromFormat.length - 1]);
+
+  const result = [];
+
+  for (let i = 0; i < fromFormat.length - 1; i++) {
+    dateObj[fromFormat[i]] = splitDate[i];
+  }
+
+  if (fromFormat.includes('YY') && toFormat.includes('YYYY')) {
+    if (dateObj.YY) {
+      if (dateObj.YY >= 30) {
+        dateObj.YYYY = '19' + dateObj.YY;
+      } else {
+        dateObj.YYYY = '20' + dateObj.YY;
+      }
+    }
+  }
+
+  if (fromFormat.includes('YYYY') && toFormat.includes('YY')) {
+    if (dateObj.YYYY) {
+      dateObj.YY = dateObj.YYYY.slice(-2);
+    }
+  }
+
+  if (dateObj.hasOwnProperty('YYYY')) {
+    dateObj.YY = dateObj.YYYY.slice(2);
+  }
+
+  for (let i = 0; i < toFormat.length - 1; i++) {
+    result.push(dateObj[toFormat[i]]);
+  }
```

Comment:
The loop iterates over `toFormat.length - 1`, which may lead to missing the last format element. Ensure that the loop covers all elements of the `toFormat` array.